vars/mtaBuild.groovy

```diff
@@ -1,16 +1,41 @@
 import com.sap.piper.Utils
 
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.ConfigurationType
+
 
 def call(Map parameters = [:]) {
 
-    handlePipelineStepErrors (stepName: 'mtaBuild', stepParameters: parameters) {
+    def stepName = 'mtaBuild'
+
+    List parameterKeys = [
+        'buildTarget',
+        'mtaJarLocation'
+    ]
+
+    List stepConfigurationKeys = [
+        'mtaJarLocation'
+    ]
+
+    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        setupCommonPipelineEnvironment script: script
```

Comment:
In my understanding this is more a cross step, which is expected to be executed once at the beginning of the pipeline. With that the configuration gets reloaded. What happens when someone in an earlier step updated the configuration. Maybe doing this is not a good pattern, but technically it is possible, since the configuration related classes are not immutable.

Reply:
Yes, good question, prepareDefaultValues then should be also done before. Which approach should we follow?

Reply:
@OliverNocon @CCFenner: What is your opinion on calling setupCommonPipelineEnviornment inside each step? As @alejandraferreirovidal outlines this is also related to the question where to call prepareDefaultValues

Reply:
I'm also not a fan of initialising the default inside each step, can't we push this to the config loader or pipelineEnv?

Reply:
IMHO we should rely on setupCommonPipelineEnvironment being called before.

Typical flow in a pipeline would be:

```
setUpCommonPipelineEnvironment ...
artifactSetVersion ...
mtaBuild ...
...
```

In addition `setupCommonPipelineEnvironment` can be called with custom parameters which would be possible overwritten.

For the other steps we don't call the setup... again ...

Reply:
Ok, yes, it makes sense, @OliverNocon what about calling 'prepareDefaultValues'? This is called also inside 'setupCommonPipelineEnvironment'